rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -0,0 +1,420 @@
+          ActivationConfig.ForBlock activations) {
+        // Create Pending federation (doing this to avoid voting the pending Federation)
+        var newFederationMembers = federationKeys.stream()
+            .map(newFederatorKey ->
+                new FederationMember(
+                    newFederatorKey.getLeft(),
+                    newFederatorKey.getMiddle(),
+                    newFederatorKey.getRight()))
+            .toList();
+        var pendingFederation = new PendingFederation(newFederationMembers);
+        // Set pending federation
+        federationStorageProvider.setPendingFederation(pendingFederation);
+        federationStorageProvider.save(BRIDGE_CONSTANTS.getBtcParams(), activations);
+
+        // Proceed with the powpeg change
+        federationSupport.commitFederation(false, pendingFederation.getHash(), bridgeEventLogger);
+
+        // Since the proposed federation is commited, it should be null in storage
+        assertNull(federationStorageProvider.getPendingFederation());
+
+        // return the new federation
+        return pendingFederation.buildFederation(
+            Instant.EPOCH,
+            0,
+            BRIDGE_CONSTANTS.getFederationConstants(),
+            activations);
+    }
+
+    private void commitProposedFederation(ActivationConfig.ForBlock activations) {
+        // Verify that the proposed federation exists in storage
+        var proposedFederation = 
+            federationStorageProvider.getProposedFederation(BRIDGE_CONSTANTS.getFederationConstants(), activations);
+        assertTrue(proposedFederation.isPresent());
+
+        // As in commitPendingFederation util method, to avoid the SVP process
+        // we will commit directly
+        federationSupport.commitProposedFederation();
+    
+        // Since the proposed federation is commited, it should be null in storage
+        proposedFederation = 
+            federationStorageProvider.getProposedFederation(BRIDGE_CONSTANTS.getFederationConstants(), activations);
+        assertTrue(proposedFederation.isEmpty());
+    }
+    
+    private void activateNewFederation(ActivationConfig.ForBlock activations) {
+        // Move the required blocks ahead for the new powpeg to become active
+        var blockNumber = 
+            initialBlock.getNumber() + BRIDGE_CONSTANTS.getFederationConstants().getFederationActivationAge(activations);
+        Block activationBlock = mock(Block.class);
+        when(activationBlock.getNumber()).thenReturn(blockNumber);
+
+        // Now the new bridgeSupport points to the new block where the new federation
+        // is considered to be active
+        bridgeSupport = getBridgeSupportFromExecutionBlock(activationBlock, activations);
+    }
+    
+    private BridgeSupport getBridgeSupportFromExecutionBlock(Block executionBlock, ActivationConfig.ForBlock activations) {
+        FederationSupport fedSupport = FederationSupportBuilder.builder()
```

Comment:
```suggestion
        federationSupport = FederationSupportBuilder.builder()
```
use the global variable?

Reply:
We need to rebuild this with the new execution block. Federation support is used as the implementation for other stuff. Check the legacy class if you want to understand the whole picture better.

Reply:
okay I see. We need the impl to commit without having to vote